Octokit/Http/SimpleJsonSerializer.cs

```diff
@@ -92,6 +92,17 @@ public override object DeserializeObject(object value, Type type)
                 {
                     if (ReflectionUtils.GetTypeInfo(type).IsEnum)
                     {
+                        //first try to get all custom attributes
+                        var fields = type.GetRuntimeFields();
+                        foreach (var field in fields)
+                        {
+                            var attribute = (ParameterAttribute)field.GetCustomAttribute(typeof(ParameterAttribute));
+                            if (attribute != null)
+                            {
+                                if (attribute.Value.Equals(value))
+                                    return field.GetValue(null);
+                            }
+                        }
```

Comment:
cool so to paraphrase/clarify you are saying we dont need to upfront cache everything, but we may as well keep a running cache of anything we've already evaluated so we dont have to do it more than once?